pkg/utils/watchChaosContainer.go

```diff
@@ -2,32 +2,57 @@ package utils
 
 import (
 	"time"
-
 	"github.com/pkg/errors"
 	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+
+var (
+	//chaosPod holds the experiment pod spec
+	chaosPod = &corev1.Pod{}
+)
+
+
+// GetChaosPodName gets the name of the chaos experiment pod launched by the runner
+func GetChaosPod(expDetails *ExperimentDetails, clients ClientSets) (*corev1.Pod, error){
+	chaosPodList, err := clients.KubeClient.CoreV1().Pods(expDetails.Namespace).List(metav1.ListOptions{LabelSelector: "job-name=" + expDetails.JobName})
+	if err != nil || len(chaosPodList.Items) == 0 {
+    	return nil, errors.Wrapf(err, "Unable to get the chaos pod, due to error: %v", err)
+	} else if len(chaosPodList.Items) > 1 {
+		// Cases where experiment pod is rescheduled by the job controller due to 
+		// issues while the older pod is still not cleaned-up
+		return nil, errors.New("Multiple pods exist with same jobname label")
```

Comment:
Can we log the name of the experiment also or some more detail for debug it easily?

Reply:
Considering we are doing a 1-1 map b/w engine and experiment, this should be easily debuggable. Even if there are more than 1 experiments listed in the engine, their execution is sequential with events showing exp that has failed.

Reply:
ok